src/lib.rs

```diff
@@ -508,6 +508,16 @@ pub trait Rng: RngCore {
     /// let mut rng = thread_rng();
     /// println!("{}", rng.gen_bool(1.0 / 3.0));
     /// ```
+    ///
+    /// # Accuracy note
+    ///
+    /// `gen_bool` uses 32 bits of the RNG, so if you use it to generate close
+    /// to or more than 2^32 results, a tiny bias may becomes noticable.
+    /// A notable consequence of the method used here is that the worst case is
+    /// `rng.gen_bool(0.0)`: it has a chance of 1 in 2^32 of being true, while
```

Comment:
Sorry, I told you to use `<=` after examining the upper bound but not the lower. I guess it's better to use `<` then and ensure `gen_bool(0.0)` is correct.

Reply:
Then we would shift to problem to `gen_bool(1.0)`. But I see our implementation as reasonable. We could alternatively just always return `false` for `gen_bool(0.0)`. But I wouldn't bother, using `gen_bool` like this is really nor very sensible (as I tried to wr3ite in the first comment, and the comment in the code).

Reply:
To top it off, some generators are explicitly designed not to produce 0. We could use `(rng.gen() - 1) < p_int`, but I guess that's slower?

Reply:
:smile: Good point. Nothing seems perfect any more once you know too many details...

That would have to use a wrapping subtraction, and would also only shift the problem to from `0.0` to `1.0`.

Reply:
No, it should use rejection sampling — but then there's no point shifting. I agree, we're thinking too much into this :laughing: